src/cloudformation_cli_python_lib/resource.py

```diff
@@ -145,15 +197,23 @@ def _parse_request(
 
     @_ensure_serialize
     def __call__(
-        self, event_data: MutableMapping[str, Any], _context: Any
+        self, event_data: MutableMapping[str, Any], context: LambdaContext
     ) -> MutableMapping[str, Any]:
         try:
             ProviderLogHandler.setup(event_data)
             parsed = self._parse_request(event_data)
             session, request, action, callback_context = parsed
-            progress_event = self._invoke_handler(
-                session, request, action, callback_context
-            )
+            invoke = True
+            while invoke:
+                progress_event = self._invoke_handler(
+                    session, request, action, callback_context
+                )
+                invoke = self.schedule_reinvocation(
+                    HandlerRequest.deserialize(event_data),
+                    progress_event,
+                    context,
+                    Credentials(**event_data["requestData"]["platformCredentials"]),
```

Comment:
might be better to do this inside `_parse_request` for the error handling, and e.g. return a `platform_session` for re-use. we'll need it anyway for async reporting the status